### PR TITLE
IGNITE-18160 Delete physical data after a Raft Group is stopped

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-18160

### CODE REVIEW

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java [hunk 1]
```diff
@@ -391,6 +435,85 @@ void testRebalanceRetryWhenCatchupFailed() throws Exception {
         assertEquals(3, getPartitionClusterNodes(2, 0).size());
     }
 
+    @Test
+    @Tag(TEST_TX_STORAGE_TAG)
```
Comment (rpuch) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java:439]: >Tags are used to filter which tests are executed for a given test plan

But here, they are used to signal that some customizations need to be applied in the `before()` method. This looks like an abuse. It seems that a custom annotation should be added instead.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 2]
```diff
@@ -2102,4 +2055,60 @@ private CompletableFuture<PartitionStorages> getOrCreatePartitionStorages(TableI
...
+                    // no-op
+                }
+
+                InternalTable internalTable = tablesByIdVv.latest().get(tableId).internalTable();
+
+                // Should be done fairly quickly.
+                allOf(
+                        internalTable.storage().destroyPartition(partitionId),
+                        runAsync(() -> internalTable.txStateStorage().destroyTxStateStorage(partitionId), ioExecutor)
+                ).join();
```
Comment (rpuch) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:2110]: It looks like we only destroy partition data, but the raft group data (log and metastorage) remain untouched. The Jira issue mentions that we need to destroy the log as well.

I think we can move the log and RAFT metastorage destruction to another issue, but we need a TODO mentioning it.

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java [hunk 3]
```diff
@@ -642,20 +815,156 @@ void stop() throws Exception {
...
+    }
+
+    private static TableDefinition createTableDefinition(String tableName) {
+        return SchemaBuilders.tableBuilder("PUBLIC", tableName).columns(
+                SchemaBuilders.column("key", ColumnType.INT64).build(),
+                SchemaBuilders.column("val", ColumnType.INT32).asNullable(true).build()
+        ).withPrimaryKey("key").build();
+    }
+
+    private void createTableForOnePartition(String tableName, int replicas, boolean testDataStorage) {
```
Comment (rpuch) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java:865]:
```suggestion
    private void createTableWithOnePartition(String tableName, int replicas, boolean testDataStorage) {
```

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java [hunk 4]
```diff
@@ -642,20 +815,156 @@ void stop() throws Exception {
...
+
+    private static @Nullable InternalTable getInternalTable(Node node, String tableName) {
+        Table table = node.tableManager.table(tableName);
+
+        assertNotNull(table, tableName);
+
+        return ((TableImpl) table).internalTable();
+    }
+
+    private static void checkInvokeDestroyPartitionStorages(Node node, String tableName, int partitionId) {
```
Comment (rpuch) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java:920]:
```suggestion
    private static void checkInvokeDestroyedPartitionStorages(Node node, String tableName, int partitionId) {
```

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java [hunk 5]
```diff
@@ -642,20 +815,156 @@ void stop() throws Exception {
...
+        InternalTable internalTable = getInternalTable(node, tableName);
+
+        verify(internalTable.storage(), times(1)).destroyPartition(eq(partitionId));
+        verify(internalTable.txStateStorage(), times(1)).destroyTxStateStorage(eq(partitionId));
```
Comment (rpuch) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java:924]:
```suggestion
        verify(internalTable.storage(), times(1)).destroyPartition(partitionId);
        verify(internalTable.txStateStorage(), times(1)).destroyTxStateStorage(partitionId);
```

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java [hunk 6]
```diff
@@ -642,20 +815,156 @@ void stop() throws Exception {
...
+    }
+
+    private static void throwExceptionOnInvokeDestroyPartitionStorages(Node node, String tableName, int partitionId) {
+        InternalTable internalTable = getInternalTable(node, tableName);
+
+        doAnswer(answer -> CompletableFuture.failedFuture(new StorageException("From test")))
+                .when(internalTable.storage())
+                .destroyPartition(eq(partitionId));
```
Comment (rpuch) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java:932]:
```suggestion
                .destroyPartition(partitionId);
```

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java [hunk 7]
```diff
@@ -642,20 +815,156 @@ void stop() throws Exception {
...
+
+        doAnswer(answer -> CompletableFuture.failedFuture(new IgniteInternalException("From test")))
+                .when(internalTable.txStateStorage())
+                .destroyTxStateStorage(eq(partitionId));
```
Comment (rpuch) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java:936]:
```suggestion
                .destroyTxStateStorage(partitionId);
```

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java (hunk 4 above)
Comment (tkalkirill) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java:920]: Fix it

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java (hunk 6 above)
Comment (tkalkirill) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java:932]: Fix it

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java (hunk 7 above)
Comment (tkalkirill) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java:936]: Fix it

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java (hunk 1 above)
Comment (tkalkirill) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java:439]: Fix it

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java (hunk 2 above)
Comment (tkalkirill) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:2110]: Added TODO

**Review by rpuch — APPROVED** [review]

**Pull request merged**